pulp_container/app/registry_api.py

```diff
@@ -475,6 +479,161 @@ def get_queryset(self, *args, **kwargs):
+                    req_label_exists.add(label)
+                else:
+                    if any(v != values[0] for v in values[1:]):
+                        unsatisfiable = True
+                        break
+                    label = key[len("label:") :]
+                    req_label_values[label] = values[0]
+            else:
+                # This covers unknown query parameters (which we consider as unsatisfiable for
+                # simplicity, rather than causing some error response) as well as any annotation:...
+                # query parameters (whose presence necessarily cause the request to be
+                # unsatisfiable, as this implementation never announces any Annotations):
+                unsatisfiable = True
+                break
+
+        results = []
+
+        if not unsatisfiable:
+            # This part copied from CatalogView:
+            queryset = models.ContainerDistribution.objects.all().only("base_path")
+            distribution_permission = "container.pull_containerdistribution"
+            namespace_permission = "container.namespace_pull_containerdistribution"
+            public_repositories = queryset.filter(private=False)
+            repositories_by_distribution = get_objects_for_user(
+                self.request.user, distribution_permission, queryset
+            )
+            namespaces = models.ContainerNamespace.objects.all()
+            repositories_by_namespace = get_objects_for_user(
+                self.request.user, namespace_permission, namespaces
+            )
+            repositories_by_namespace = queryset.filter(namespace__in=repositories_by_namespace)
+            accessible_repositories = repositories_by_distribution & repositories_by_namespace
+            repositories = (public_repositories | accessible_repositories).distinct()
+
+            if req_repository is not None:
+                repositories = repositories.filter(base_path=req_repository)
+
+            for distribution in repositories:
+                images = []
+                if distribution.repository:
+                    repository_version = distribution.repository.latest_version()
+                elif distribution.repository_version:
+                    repository_version = distribution.repository_version
+                tags = models.Tag.objects.filter(pk__in=repository_version.content)
+                manifests = models.Manifest.objects.filter(
+                    pk__in=repository_version.content,
+                    media_type__in=(models.MEDIA_TYPE.MANIFEST_V2, models.MEDIA_TYPE.MANIFEST_OCI),
+                )
+                for manifest in manifests:
+                    with storage.open(manifest.config_blob._artifacts.get().file.name) as file:
+                        raw_data = file.read()
+                    config_data = json.loads(raw_data)
+                    labels = config_data.get("config", {}).get("Labels")
```

Comment:
config is required for these media_types

Reply:
Sorry, I don't understand what you mean with that.

Reply:
i misread, ignore